### PR TITLE
send patient country to RS

### LINKED ISSUE #3374: Send country to ReportStream
### Why are we doing this?
Vermont has requested patient country as they have some patients from Canada. We store patient country, but are not currently passing the value along to ReportStream.

### Action requested
Update [`TestEventExport`](https://github.com/CDCgov/prime-simplereport/blob/7f0c58ec6b376d5f26763e7de1dca1745f3752de/backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java#L38-L38) to pass patient country information to ReportStream.

We'll need to make sure null countries default to USA.

### Additional context
https://github.com/CDCgov/prime-reportstream/issues/3848#issuecomment-1034040192

### PR DESCRIPTION
**Pull request by zdeveloper**
## Related Issue or Background Info

- fixed #3374

## Changes Proposed

- send country field to report stream
- `Patient_country` is the new field name

### Infrastructure
- [X] **Consult the results of the `terraform-plan` job inside the "Terraform Checks" workflow run for this PR. Confirm that there are no unexpected changes!**

### Design
- [X] Any UI/UX changes have a designer as a reviewer, and changes have been approved
- [X] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke-tested by both the engineering and design teams

### Content
- [X] Any content changes (including new error messages) have been approved by content team

### Support
- [X] Any changes that might generate new support requests have been flagged to the support team
- [X] Any changes to support infrastructure have been demo'd to support team

### Testing
- [X] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [X] Database changes are submitted as a separate PR
  - [X] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [X] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [X] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
  - [X] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [X] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [X] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [X] Any dependencies introduced have been vetted and discussed

## Cloud
- [X] DevOps team has been notified if PR requires ops support
- [X] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

File: backend/src/test/java/gov/cdc/usds/simplereport/api/model/TestEventExportTest.java [hunk 1]
```diff
@@ -194,4 +194,30 @@ void sendCorrectionReason() {
...
+    Facility facility = _dataFactory.createValidFacility(org);
+    Person person = _dataFactory.createFullPerson(org);
+    TestEvent testEvent =
+        _dataFactory.createTestEvent(person, facility, TestResult.NEGATIVE, false);
+
+    // WHEN
+    TestEventExport exportedEvent = new TestEventExport(testEvent);
+
+    // THEN
+    assertEquals("Fred", exportedEvent.getPatientFirstName());
```
Comment (emmastephenson) [inline backend/src/test/java/gov/cdc/usds/simplereport/api/model/TestEventExportTest.java:211]: Could you also add tests to confirm that this works with a null country and with a non-US country?

**Review by emmastephenson — COMMENTED** [review]
Overall LGTM, just a question about the behavior with a null country.

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java [hunk 2]
```diff
@@ -201,6 +201,11 @@ public String getPatientZipCode() {
     return patient.map(Person::getZipCode).orElse(null);
   }
 
+  @JsonProperty("Patient_country")
+  public String getPatientCountry() {
+    return patient.map(Person::getCountry).orElse(USA);
```
[the remark this answers is not in the document]
Comment (zdeveloper) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java:206]: `null`, we shouldn't have empty country fields in the DB. per this code 
https://github.com/CDCgov/prime-simplereport/blob/81debeb8889a52052b58467d91e7b4c1c34a2d35/backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientMutationResolver.java#L106

File: backend/src/test/java/gov/cdc/usds/simplereport/api/model/TestEventExportTest.java (hunk 1 above)
Comment (zdeveloper) [reply backend/src/test/java/gov/cdc/usds/simplereport/api/model/TestEventExportTest.java:211]: good call, fixed in https://github.com/CDCgov/prime-simplereport/pull/3392/commits/9e36464892ea2f6f2ed4a8f078998abdc488518a

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (emmastephenson) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java:206]: That's true for any new patients, but patients created before we added country last Nov. can still be null. We have code on the frontend to read those `nulls` as `USA`, but they're still null in the database: https://github.com/CDCgov/prime-simplereport/blob/15f9113dbb7a240b2ce3709a29c8e83cfb26823d/frontend/src/app/patients/Components/PersonForm.tsx#L103

[the remark this answers is not in the document]
Comment (zdeveloper) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java:206]: yes, hence why the `null` check

**Review by emmastephenson — APPROVED** [review]
LGTM with the new tests, thanks!

**Review by nickclyde — DISMISSED** [review]
Thanks for adding the null test, LGTM

**Review by nickclyde — APPROVED** [review]
